cargo-geiger/src/cli.rs

```diff
@@ -113,19 +105,17 @@ pub fn get_workspace(
 }
 
 pub fn resolve<'a, 'cfg>(
+    args: &Args,
     package_id: PackageId,
     registry: &mut PackageRegistry<'cfg>,
     workspace: &'a Workspace<'cfg>,
-    features: &[String],
-    all_features: bool,
-    no_default_features: bool,
 ) -> CargoResult<(PackageSet<'a>, Resolve)> {
     let dev_deps = true; // TODO: Review this.
-    let uses_default_features = !no_default_features;
+    let uses_default_features = !args.no_default_features;
     let opts = ResolveOpts::new(
         dev_deps,
-        features,
-        all_features,
+        &args.features.clone(),
+        args.all_features,
         uses_default_features,
     );
     let prev = ops::load_pkg_lockfile(workspace)?;
```

Comment:
I like the idea to bundle up some of the parameters into a struct here, however `Args` provides around 30 members of which only 3 are needed by this function. This is probably of lesser importance but I would like us to create a new struct for this purpose, to avoid introducing tight coupling between all of the command line arguments and this function.

Reply:
Hi @anderejd, that's a good shout, we definitely don't need to be throwing the whole `Args` struct about.
I've update the PR to pull out a struct just for these values, which should allow a decoupling from the remainder of the command line arguments.
Please let me know if you had something different in mind?
Thanks,
Josh

Reply:
LGTM!